pkg/apis/pipeline/v1beta1/result_types.go

```diff
@@ -39,7 +39,7 @@ type TaskRunResult struct {
 	Type ResultsType `json:"type,omitempty"`
 
 	// Value the given value of the result
-	Value string `json:"value"`
+	Value ArrayOrString `json:"value"`
```

Comment:
This looks like a API change to me. It seems like we need to follow https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md#alpha-beta-and-ga for adding new feature and deprecation rules. Pinging @[dibyom](https://github.com/dibyom) for confirmation. This is a field that users can specify. I think we should add a separate field for the new ArrayOrString and implement on the new field

Reply:
It is a change, but it won't have effect on users and introduce breaking changes. And our TEP[0076](https://github.com/tektoncd/community/blob/main/teps/0076-array-result-types.md) requires us to change this `Value` I believe?

Reply:
We came to a conclusion that this won't be breaking the API change policy. However, we definitely are blocked by the design about whether we want to have a brand new struct on this or just rename StringOrArray.

Reply:
Ok but it doesn't block this PR right, I'm still using StringOrArray. We need to figure out that in Chuang's

Reply:
If we end up introducing a new struct, the golang library will be broken twice. Consider not making this particular line of change and be able to test everything else.

Reply:
Ok

Reply:
This thread can be closed, as we confirmed that this won't be blocked by introducing a new struct for ArrayOrString, and this is not a API change.

Reply:
sure

Reply:
I ran `tkn` which failed with:

```
tkn taskrun list
Error: failed to list TaskRuns from namespace default: unrecognized type: string
```

I haven't ran any troubleshooting on the CLI to confirm if its caused by this change 🙃 .

```
tkn version                                                                                       
Client version: 0.23.1
Pipeline version: devel
```

It might be safe to merge this PR but the CLI might will have to updated before adopting the latest pipelines release (with these changes). @vdemeester @vinamra28

Reply:
Thanks for checking! And we also need to change that type for Pipelinerun result in the following PR.

Reply:
Yeah sure!

Reply:
cc:@ywluogg so we should send emails for this?

Reply:
It looks like api mismatch between the pipeline and cli, it should be fixed when we release this change and then bump the pipeline version for cli? Am I correct?

Reply:
/cc @tektoncd/cli-maintainers

Reply:
> It looks like api mismatch between the pipeline and cli, it should be fixed when we release this change and then bump the pipeline version for cli? Am I correct? @pritidesai

I think It will be fixed when CLI bumps the pipelines version (which will have these changes) and update their usage/implementation to change the type of the results from `string` to `StringOrArray`. The error reported here is coming from:

```
./pkg/cmd/taskrun/list.go:                              return fmt.Errorf("failed to list TaskRuns from namespace %s: %v", p.Namespace(), err)
```